File: src/types/PromptModel.ts
// Interfaces for PromptModel creation and retrieval processes

/**
 * Interface for PromptModel creation request.
 */
export interface CreatePromptModelRequest {
  project_uuid: string;
  name: string;
}

/**
 * Interface for the request to edit a PromptModel's information.
 */
export interface EditPromptModelRequest {
  uuid: string;
  name: string;
}

/**
 * Interface for the request to delete a PromptModel.
 */
export interface DeletePromptModelRequest {
  uuid: string;
}

/**
 * Interface for the request to read an project's PromptModels.
 */
export interface ReadPromptModelsRequest {
  project_uuid: string;
}

/**
 * General PromptModel interface for representing a PromptModel in the system.
 */
export interface PromptModel {
  id: number;
  created_at: string;
  uuid: string;
  name: string;
  online: boolean;
  project_uuid: string;
}

File: api/web/models/prompt_model.py
"""APIs for PromptModel"""
from modules.types import PMObject


class PromptModelInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    name: str
    online: bool

    project_uuid: str


class CreatePromptModelBody(PMObject):
    name: str
    project_uuid: str
